Tensile/TensileCreateLibrary.py

```diff
@@ -484,30 +487,32 @@ def prepAsm(
     os.chmod(assemblerFileName, 0o777)
 
 
-################################################################################
-def buildKernelSourceAndHeaderFiles(results, outputPath):
-    """
-    Logs errors and writes appropriate info to kernelSourceFile and kernelHeaderFile.
+def buildKernelSourceAndHeaderFiles(
+    results: ProcKernResult, outputPath, lazyLoading: bool, mergeFiles: bool, numMergedFiles: int
+) -> Tuple[List[str], Dict[str, int]]:
+    """Logs errors and writes appropriate info to kernel source and header files.
 
     Arguments:
-      results:              list of (err, src, header, kernelName, filename)
-      outputPath:           path to source directory
-      kernelsWithBuildErrs: Dictionary to be updated with kernels that have errors
-      kernelSourceFile:     File to write source data to
-      kernelHeaderFile:     File to write header data to
+        results: List of (err, src, header, kernelName, filename)
```

Comment:
The name `results` may confuse. It suggests to me that this is the result of the function rather than solely input.

Reply:
I did not change this, but instead added sematic type-hints to the functions to help guide the developer.